### PR TITLE
Bug 1547061 - Port FXA signup form to trailhead first run

### PR DESCRIPTION
**Pull request by rlr**

### CODE REVIEW

File: content-src/asrouter/templates/Trailhead/Trailhead.jsx [hunk 1]
```diff
@@ -15,26 +16,83 @@ const FLUENT_FILES = [
...
+      emailInput: "",
+      isModalOpen: true,
+      flowId: "",
+      flowBeginTime: 0,
+    };
+    this.didFetch = false;
   }
 
-  componentWillMount() {
+  async componentWillMount() {
```
Comment (rlr) [inline content-src/asrouter/templates/Trailhead/Trailhead.jsx]: is it ok to make this async? Is the rendering going to be delayed until the fetch happens below?

Comment (rlr) [reply content-src/asrouter/templates/Trailhead/Trailhead.jsx]: I originally had it in `componentDidMount` but you aren't supposed to call `setState` from there.

File: content-src/asrouter/templates/Trailhead/Trailhead.jsx [hunk 2]
```diff
@@ -70,15 +128,25 @@ export class Trailhead extends React.PureComponent {
...
-              href="https://accounts.firefox.com/legal/privacy" />
-          </p>
-          <button data-l10n-id="onboarding-join-form-continue" />
+          <form method="get" action={this.props.fxaEndpoint} target="_blank" rel="noopener noreferrer" onSubmit={this.onSubmit}>
+            <input name="service" type="hidden" value="sync" />
+            <input name="action" type="hidden" value="email" />
+            <input name="context" type="hidden" value="fx_desktop_v3" />
+            <input name="entrypoint" type="hidden" value="activity-stream-firstrun" />
+            <input name="utm_source" type="hidden" value="activity-stream" />
+            <input name="utm_campaign" type="hidden" value="firstrun" />
```
Comment (rlr) [inline content-src/asrouter/templates/Trailhead/Trailhead.jsx]: I wonder if the FXA team wants new values for any of these params.

Comment (rlr) [reply content-src/asrouter/templates/Trailhead/Trailhead.jsx]: Per the copy doc, I added utm_term=trailhead so the FxA folks can filter the data easily.

File: test/unit/asrouter/templates/Trailhead.test.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import {actionCreators as ac, actionTypes as at} from "common/Actions.jsm";
+import {mountWithIntl} from "test/unit/utils";
```
Comment (k88hudson) [inline test/unit/asrouter/templates/Trailhead.test.jsx]: try just using `mount`

**Pull request merged**